qiskit_nature/operators/second_quantization/qubit_converter.py

```diff
@@ -40,58 +106,235 @@ def z2symmetry_reduction(self) -> Optional[List[int]]:
+                                 format(z2symmetry_reduction))
+
         self._z2symmetry_reduction = z2symmetry_reduction
-        self._z2symmetries = None
+        self._invalidate()
+
+    @property
+    def did_two_qubit_reduction(self) -> bool:
+        """Get two qubit reduction as used from conversion
+
+        Raises:
+            QiskitNatureError: If property is accessed before a successful conversion
+        """
+        self._check_valid()
+        return self._did_two_qubit_reduction
 
     @property
-    def z2symmetries(self) -> Optional[Z2Symmetries]:
-        """Get z2symmetries. Will be `None` until :meth:`to_qubit_ops` has been called."""
-        return self._z2symmetries
+    def num_particles(self) -> Optional[Tuple[int, int]]:
+        """Get the number of particles as supplied to conversion
 
-    def to_qubit_ops(self, second_q_ops: List[SecondQuantizedOp],
-                     z2symmetry_reduction: List[int] = None,
-                     ) -> List[Optional[PauliSumOp]]:
+        Raises:
+            QiskitNatureError: If property is accessed before a successful conversion
+        """
+        self._check_valid()
+        return self._num_particles
+
+    @property
+    def z2_symmetries(self) -> Z2Symmetries:
+        """Get z2_symmetries as detected from conversion. This may indicate no symmetries
+        if none were detected.
+
+        Raises:
+            QiskitNatureError: If property is accessed before a successful conversion
+        """
+        self._check_valid()
+        return copy.deepcopy(self._z2symmetries)
+
+    def convert(self, second_q_ops: Union[SecondQuantizedOp, List[SecondQuantizedOp]],
+                num_particles: Optional[Tuple[int, int]] = None,
+                z2symmetry_reduction: Optional[List[int]] = None,
+                ) -> Union[PauliSumOp, List[Optional[PauliSumOp]]]:
         """
         Maps the given list of second quantized operators to qubit operators. Also it will
         carry out z2 symmetry reduction on the qubit operators if z2symmetry_reduction has
-        been specified.
+        been specified. For convenience a single operator may be passed, without wrapping
+        it in a List, and in which case a single qubit operator will likewise be returned.
 
         Args:
-            second_q_ops: The list of second quantized operators to be converted
+            second_q_ops: A second quantized operator, or list thereof to be converted.
+            num_particles: Needed for two qubit reduction to determine correct sector. If
+                not supplied, even if two_qubit_reduction is possible, it will not be done.
             z2symmetry_reduction: Optional z2symmetry reduction, the sector of the symmetry
 
         Returns:
-            A list of qubit operators or the same length as the second_q_ops list. The first
+            A qubit operator or list of thereof the same length as the second_q_ops list. The first
```

Comment:
```suggestion
            A qubit operator or list thereof of the same length as the second_q_ops list. The first
```

Reply:
Yes, it maybe the comments need updating. When I am done with the testing and reviewing that I will fix things up. As it is the grammar there is messed up as I already made one change